tests/export_test.py

```diff
@@ -65,3 +65,20 @@ def test_transcribe_from_wav():
     assert m.transcribe_from_file(
         ROOT.joinpath("samples", "jfk.wav").resolve().__fspath__()
     ) == m.transcribe(preprocess(ROOT / "samples" / "jfk.wav"))
+
+
+def test_callback():
+    m = Whisper.from_pretrained("tiny.en")
+
+    def handleNewSegment(n_new):
+        segment = m.context.full_n_segments() - n_new
+        while segment < m.context.full_n_segments():
+            handleNewSegment.text += m.context.full_get_segment_text(segment)
+            segment += 1
```

Comment:
I would prefer exposing the callback to match the arguments provided from whisper.cpp

So users can do the following

```python
def new_segment_callback(context: Context, n_new: int, userdata: t.Any):
    ...
```

having to define the model instance first feels a bit weird.

Reply:
Right, getting `Context` makes sense to me. Not sure I see many usecases for `userdata` though.

Reply:
For userdata, user can defined how that would be via `set_new_segment_callback_user_data` right?

Reply:
One usecase I can think of is when you need to process stereo audio, you can pass those via user_data.

Another one is for supporting diarization and VAD.

Reply:
> For userdata, user can defined how that would be via `set_new_segment_callback_user_data` right?

Well not really, the `set_new_segment_callback_user_data` does no longer exists in this PR. 
As far as I know, the `void * user` style of passing user data to callback functions is the "C"-way of doing stuff and when using `C++` usually something like `std::function`, which allows stuff from the environment to be captured (ie you can create a `std::function` from a lambda). That way one does not need the `void * user` pointer and has type safety (as the user data does not have to be casted to a `void *` in this instance.